python-package/xlearn/sklearn.py

```diff
@@ -0,0 +1,185 @@
+        self._convert_data(X, y, temp_train_file.name)
+        self._XLearnModel.setTrain(temp_train_file.name)
+
+        # fit model
+        self._XLearnModel.fit(params, self._temp_model_file.name)
+
+        # remove temporary files for training
+        self._remove_temp_file(temp_train_file)
+
+    def predict(self, X):
+        """ Generate prediction using feature matrix X
+
+        :param X: array-like
+                  Feature matrix
+        :return: prediction
+        """
+        # convert data to libsvm format
+        temp_test_file = tempfile.NamedTemporaryFile(delete=True)
+        self._convert_data(X, None, temp_test_file.name)
+        self._XLearnModel.setTest(temp_test_file.name)
+
+        # generate output
+        temp_output_file = tempfile.NamedTemporaryFile(delete=True)
+        self.get_model().predict(self._temp_model_file.name, temp_output_file.name)
+
+        # read output into numpy
+        pred = np.loadtxt(temp_output_file.name)
+
+        # remove temporary test data and output file
+        self._remove_temp_file(temp_test_file)
+        self._remove_temp_file(temp_output_file)
+
+        return pred
+
+    def feature_importance_(self):
+        """TODO: analyze weight matrix to get feature importance"""
+        pass
+
+    def _convert_data(self, X, y, filepath):
+        """
+        Convert feature matrix X and label y into libsvm format
+        :param X: feature matrix
+        :param y: label or None
+        """
+        if y is None:
+            # create dummy label for test
+            y = np.zeros(X.shape[0], dtype=np.int8)
+
+        try:
+            dump_svmlight_file(X, y, filepath)
+        except:
+            raise Exception('failed to convert feature matrix X and label y to libsvm format')
+
+    def _remove_temp_file(self, temp_file):
+        # The temp_file might be converted to binary file during training/inference.
+        # remove both original temp_file and derived binary file if exist
+        temp_bin_file = temp_file.name + '.bin'
+        if os.path.exists(temp_bin_file):
+            os.remove(temp_bin_file)
+        temp_file.close()
+
+    def __delete__(self, instance):
+        self._temp_model_file.close()
+
+
+class FMModel(BaseXLearnModel):
+    def __init__(self, model_type='fm', task='binary', metric='auc',
+                        lr=0.2, k =4, reg_lambda=0.1, init=0.1, fold=1, epoch=5,
+                        opt='sgd', nthread=4, alpha=1, beta=1, lambda_1=1, lambda_2=1,
+                        **kwargs):
+        assert model_type == 'fm', 'Model type must be set to fm for FMModel'
+        super(FMModel, self).__init__(model_type, task, metric,
+                                      lr, k, reg_lambda, init, fold, epoch,
+                                      opt, nthread, alpha, beta, lambda_1, lambda_2,
+                                      **kwargs)
+
+    def __delete__(self, instance):
+        super(FMModel, self).__delete(instance)
```

Comment:
Could you please add the other model (LR and FFM)?

Reply:
Sure, I can add LR and FFM in this week